internal/provider/data_source_aws_network_peering.go

```diff
@@ -107,31 +108,63 @@ func dataSourceAwsNetworkPeeringRead(ctx context.Context, d *schema.ResourceData
 	peeringID := d.Get("peering_id").(string)
 	waitForActive := d.Get("wait_for_active_state").(bool)
 
+	// Query for the peering.
 	log.Printf("[INFO] Reading network peering (%s)", peeringID)
 	peering, err := clients.GetPeeringByID(ctx, client, peeringID, hvnID, loc)
 	if err != nil {
 		return diag.Errorf("unable to retrieve network peering (%s): %v", peeringID, err)
 	}
 
-	if waitForActive && peering.State != networkmodels.HashicorpCloudNetwork20200907PeeringStateACTIVE {
-		peering, err = clients.WaitForPeeringToBeActive(ctx, client, peering.ID, hvnID, loc, peeringCreateTimeout)
-		if err != nil {
-			return diag.FromErr(err)
-		}
+	// Set the globally unique id of this peering in state.
+	link := newLink(peering.Hvn.Location, PeeringResourceType, peering.ID)
+	url, err := linkURL(link)
+	if err != nil {
+		return diag.FromErr(err)
 	}
+	d.SetId(url)
 
-	// Network peering found, update resource data.
+	// Store resource data.
 	if err := setAwsPeeringResourceData(d, peering); err != nil {
 		return diag.FromErr(err)
 	}
 
-	// Set the globally unique id of this peering in the state.
-	link := newLink(peering.Hvn.Location, PeeringResourceType, peering.ID)
-	url, err := linkURL(link)
-	if err != nil {
-		return diag.FromErr(err)
+	// Skip waiting.
+	if !waitForActive || peering.State == models.HashicorpCloudNetwork20200907PeeringStateACTIVE {
+		return nil
 	}
-	d.SetId(url)
 
-	return nil
+	// If it's in a state where it could later become ACTIVE, wait.
+	waitState := false
+	for _, state := range clients.WaitForPeeringToBeActiveStates {
```

Comment:
Switch statement? This looks awkward.

Reply:
I started with that but was then re-defining the states we already list in a slice in the peering clients package, which was more awkward imo